src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,55 @@
+import classnames from 'classnames';
+import { px } from '../../utils';
+import CloseIcon from '../../assets/close.svg?react';
+import Button from '../Button/Button';
+
+export interface ModalProps extends React.HTMLAttributes<HTMLElement> {
+  /**
+   * Boolean that will determine if the modal is open
+   */
+  open?: boolean;
+  /**
+   * Function to close the modal
+   */
+  onClose?: () => void;
+}
+
+const onClosePlaceholder = () => {
+  console.log('On close clicked');
+};
+
+/**
+ * ## Overview
+ *
+ * A component for displaying a modal.
+ *
+ */
+
+const Modal = ({ children, open = false, onClose = onClosePlaceholder, ...props }: ModalProps) => {
+  if (!open) {
+    return null;
+  }
+
+  const buttonClick = (e: React.MouseEvent) => {
+    e.stopPropagation();
+    onClose();
+  };
+
+  return (
+    <div className={classnames(`${px}-modal`)} onClick={onClose} data-testid="modal-background">
```

Comment:
I don't think this passes a11y as it's not setting aria stuff correctly.  Have you considered using this library https://www.radix-ui.com/primitives/docs/components/dialog to get the accessibility right?  Or if that seems like overkill this one which is lighter weight? https://reactcommunity.org/react-modal/